### ISSUE TITLE
[Feature] Oxen Monetization: Integration of Lokinet Marketplace Referral Program in Session

### ISSUE THREAD

**Issue by venezuela01** [oxen-io/session-desktop#2813]
### Is there an existing request for feature?

- [X] I have searched the existing issues

### What feature would you like?

# Oxen Monetization: Integration of Lokinet Marketplace Referral Program in Session

According to official report, Session currently has about 700,000 monthly active users, and is projected to reach to 1 million by the year's end. However, it is often perceived that Session users do not significantly contribute to the value of Oxen tokens since the existing protocol does not mandate Oxen token consumption by Session users.

This proposal introduces an different perspective, advocating that Session could aid Lokinet VPN providers in accruing users. In reciprocation, the VPN vendors would pay commissions to OPTF, who would then purchase and burn Oxen tokens from the market.

## Motivation

The primary roles in this referral program are the user, the VPN vendor, and OPTF. For the successful execution of this program, it is crucial to maintain integrity among all parties and to establish a scenario beneficial to everyone, thereby incentivizing participation.

With a continuously growing number of monthly active users, the Session app presents a valuable channel of traffic. Considering the costly nature of customer acquisition and high churn rates in the VPN industry, Session's introduction of new users to VPN vendors could be highly beneficial. Session users, as a privacy-conscious group, are likely to be potential VPN customers as well. With numerous VPN vendors in the market, users might have difficulty choosing a reliable one. If Session carefully selects reputable vendors for the users, and leverages Lokinet technology to enhance privacy, it is likely to benefit users as well. If successfully referred, a modest commission could be charged to VPN vendors for every customer introduced by Session.

## Challenges and Solutions

The referral program faces several hurdles:

1. We must ascertain that VPN vendors remit the commission to OPTF whenever Session introduces a new, paying customer.
2. In cases where a customer visits the VPN vendor's website but does not finalize the purchase, OPTF should not impose a commission. Moreover, we must create a mechanism to prevent potential dishonesty by OPTF and eliminate the possibility of dispute between OPTF and VPN vendors.

Our proposal addresses these issues:

Initially, OPTF will partner with selected VPN vendors. Each vendor creates an Oxen wallet, retaining the spending key and sharing the view key with OPTF. This setup allows OPTF to monitor all incoming transactions, ensuring transparency in order completion. Alternatively, OPTF can aid VPN vendors in operating a BTCPay server, which accepts BTC and other cryptocurrencies. In this scenario, no view key is needed, as the transparency of BTC blockchain permits the tracing of incoming payments by default.

We propose to integrate a "Marketplace Community" into the Session app, similar to existing Session Communities. This channel will enable subscribed users to receive the latest offers anonymously. Each offer will detail the vendor, VPN service specifications, pricing, and promotions such as 5% or 10% discounts. The order will include an Oxen payment address or BTCPay address agreed upon by the VPN vendor and OPTF.

Users can take advantage of these offers by paying to the embedded address to claim the stated discounts. This arrangement mirrors common strategies used by internet influencers for product promotion and commission charging. They utilize promotion codes to incentivize users to purchase with a specific tracking string. In return, the promoter can keep track of successful transactions and charge commissions propotionally. In our case, the usage of onion routing and cryptocurrencies provides better user privacy protection while still keep track of order completions.

If a user finalizes a transaction and pays for VPN service through the offer's embedded crypto address, OPTF can track the transaction amount via the Oxen wallet view key or the BTC blockchain. Subsequently, OPTF can demand a commission percentage from the VPN vendor, potentially on a monthly basis. If a vendor fails to pay, OPTF can revise the Marketplace Community and excise that vendor's offers.

There is an edge case where a user might be dissatisfied with the VPN service and ask for a refund, and the VPN vendor agrees to it. In this case, OPTF shouldn't charge a commission. However, enforcing this on a protocol level is challenging, or at least difficult, without compromising user privacy. A general workaround is to trust vendors to report any refund orders, and to benchmark different vendors on refund rates. If a vendor has a significantly higher refund rate compared to the industry average, we should terminate the partnership with this vendor. The reason could be that they are either offering low-quality service with a high complaint rate or cheating on commission. We don’t know the exact case, but we can’t continue partnering with them regardless of each case.

## Anticipated Impact

Assuming a 1% conversion rate from 1 million users and a $10 monthly VPN subscription, the total transaction volume per year would equate to $1.2 million. If OPTF imposes a 10% commission on VPN vendors, this would result in an annual income of $120,000. This revenue could offset more than half of the total Oxen service network infrastructure fees and provide substantial price support. As the user base grows, more Oxen will be burned, leading to a price increase that would incentivize investors to continue supporting the Oxen project, thereby paving the way for further monetization strategies.

This strategy doesn't necessitate intricate technological solutions and can be smoothly executed using a lean startup style of trial-and-error. We could select a few VPN partners for initial experimentation, possibly implementing it on a small scale for a user subset. If successful, it will immediately generate revenues, and if the business model is validated, we could scale it up.

## Conclusion

By implementing a Lokinet Marketplace Referral Program in the Session app, we create mutual benefits for all three parties:

1. **User**: Through the proposed referral program, users gain access to carefully selected VPN brands and promotional offers. This enhances their online privacy and security. Additionally, increased revenue from the referral program could be invested in developing further improvements in the Session app, which would benefit a wider range of Session users in the long term.

2. **VPN Vendor**: VPN vendors benefit from increased customer acquisition with potentially lower acquisition costs compared to traditional marketing methods. The growing user base of the Session app presents a substantial market to tap into, reducing churn rates and boosting revenue.

3. **Oxen Stakeholders**: The OPTF receives a revenue stream through commissions paid by VPN vendors for customer referrals. This income can be used to purchase and burn Oxen, significantly offsetting the infrastructure fees of the Oxen service network and benefiting all Oxen token holders in the community. As a result, it will create price support that incentivizes investors to continue supporting the project.

**Comment by KeeJef** [comment]
Yeah i generally like this idea, we will already be looking at getting some commercial VPN providers onboarded into Lokinet as Exit providers. I think it could be cool to promote those providers products via Session. I think the conversion percentages and pricing points given here are slightly optimistic. But this could still result in a non negligible income for the network.

**Comment by venezuela01** [comment]
> Yeah i generally like this idea, we will already be looking at getting some commercial VPN providers onboarded into Lokinet as Exit providers. I think it could be cool to promote those providers products via Session. I think the conversion percentages and pricing points given here are slightly optimistic. But this could still result in a non negligible income for the network.

From my personal experience in a previous industry selling physical products, partners offered us a commission ranging from 0% to over 10%. The weighted average hovered around 8%. The actual commission rate was negotiated on a case-by-case basis. Factors like brand strength played a key role; if a partner boasted robust brand recognition, we were keen to onboard them onto our platform even without commission. On the other hand, lesser-known partners, whom we assisted in acquiring customers, were levied a higher commission.

This strategy can be mirrored for the Session Lokinet Referral Program, with a lean startup style approach.

Initially, we can partner with emerging brands, prioritizing those businesses owned by loyal Session/Lokinet community supporters like [PrivacyProShop](https://www.privacyproshop.com/) or [Rob Braxman Store](https://brax.me/prod/host.php?f=_store&h=rob&p=&version=).

* PrivacyProShop: Lokinet VPN
* Rob Braxman Store: Bytz VPN is a stealth VPN service with an option for routing through the TOR Relay.

By fostering their growth, we position ourselves to request a relatively higher commission rate, all while refining our product through hands-on experience. Once our Exit Market prototype is firmly established, we can pivot to collaborating with mainstream brands, potentially at lower commission rates.

The digital product markets usually have a higher commission rate than physical product markets. According to [Affiliate Marketing Commission Rates](https://referralrock.com/blog/affiliate-commission-rates/#h.7xicyrgoj3hz):
> Percentage commissions are most common in affiliate programs, and the average affiliate commission rate is between 5–30%.

**Comment by BeholdersEye** [comment]
Bundled advertising seems like something most users want. Unlikely to result in a fork. 🙄

**Comment by venezuela01** [comment]
Update: a list of 40+ VPN service providers who supports crypto payment.

These VPN service providers might be our potential partner for Lokinet integrated VPN.

https://cryptwerk.com/tag/vpn/

Off-topic:
For people want to run Oxen service node and pay in crypto, there is a list of 500+ hosting vendors as well:
https://cryptwerk.com/companies/hosting/
